### DISCUSSION TITLE
GitHub mobile doesn’t protect against committing directly to protected branch

### DISCUSSION THREAD

**Discussion by devinmatte** [community/community#69744] [Mobile]
When writing a commit on mobile to edit a file, the app doesn't notify you you're about to commit directly to a protected branch. It just makes the commit.

This feels potentially dangerous

<details>
<summary>ℹ️ Additional Debug Info</summary>
GitHub for iOS, Version 1.132.0 (110686114)
Device: iPad Air 4 (iOS 17.0.3)
</details>

**Comment by esahmed21** [comment]
Hi @devinmatte 👋

Thank you for your feedback! We appreciate your input. To help us investigate this issue further and provide a better solution, could you please provide some additional details? It will be great to know if you have admin privileges in the repository or which permission you’ve? Currently, we only show the option of pushing to a protected branch if a user has sufficient permissions to do otherwise the protected branch shouldn’t appear.

**Reply by devinmatte** [reply]
Yes I have permissions to override this, but I expected to see a checkbox to use my admin permission or a next view to select creating a new branch or direct commit like it does on web

**Reply by esahmed21** [reply]
Hi @devinmatte 
That makes totally sense. On web we show a red warning that user will be Bypassing rules if he committed to protected branch. Probably we need to introduce similar behaviour on mobile as well.


> or a next view to select creating a new branch or direct commit like it does on web

Regarding this point, there should be a a view in the bottom left where you can already select creating diffrente branch. i want to make sure you see that option, let me know if it's not working like that.

Thank you for this valuable feedback!
